ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -3528,6 +3528,68 @@ IHqlExpression * ThorHqlTransformer::normalizeMergeAggregate(IHqlExpression * ex
 }
 
 
+static bool reorderAggregateFields(HqlExprArray & aggregateFields, HqlExprArray & aggregateAssigns)
+{
+    bool needToReorder = false;
+    bool variableOffset = false;
+    bool dynamicOffset = false;
+    ForEachItemIn(i, aggregateFields)
+    {
+        IHqlExpression & cur = aggregateFields.item(i);
+        IHqlExpression * value = aggregateAssigns.item(i).queryChild(1);
+        if (value->isGroupAggregateFunction())
+        {
+            if (dynamicOffset)
+                needToReorder = true;
+        }
+
+        if (isUnknownSize(&cur))
+        {
+            if (value->isGroupAggregateFunction())
+                dynamicOffset = true;
+            else
+                variableOffset = true;
```

Comment:
c.f. above comment regarding symmetry of logic with code in ecl/hqlcpp/hqlhtcpp.cpp:12991.